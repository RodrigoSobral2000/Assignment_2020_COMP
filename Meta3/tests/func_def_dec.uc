void A(void);

void A(void a){
	int x;
}

int B(int a){
	int x;
	x=2;
	a=3;
}

int B(int);


int C(int);
int C(int);

int C(int a){}


int D(int);
int D(int);

int D(int a){}
int D(int a){}



int E(int);

int E(int a){}
int E(double b){}
double E(int a){}


int F(double);
int F(int a){}



int G(short a);
short G(int){
	int y, v;
	return v;
}

int H(short a);
int H(short a, short b);


int I(int,int);
int I(int);

int J(short a) {
	double d, b;
	return b;
}
int J(short a, short b){
	int a, b;
	return b;
}


int K(short a, short b){
	int o, p;
	return p;
}
int K(short a){
	int t, r;
	return t;
}