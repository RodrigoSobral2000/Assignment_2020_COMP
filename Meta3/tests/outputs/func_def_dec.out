Line 3, col 8: Invalid use of void type in declaration
Line 26, col 5: Symbol D already defined
Line 33, col 5: Symbol E already defined
Line 34, col 8: Symbol E already defined
Line 38, col 5: Conflicting types (got int(int), expected int(double))
Line 43, col 7: Conflicting types (got short(int), expected int(short))
Line 49, col 5: Conflicting types (got int(short,short), expected int(short))
Line 53, col 5: Conflicting types (got int(int), expected int(int,int))
Line 57, col 9: Conflicting types (got double, expected int)
Line 59, col 5: Symbol J already defined
Line 60, col 6: Symbol a already defined
Line 60, col 9: Symbol b already defined
Line 69, col 5: Symbol K already defined
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
A	void(void)
B	int(int)
C	int(int)
D	int(int)
E	int(int)
F	int(double)
G	int(short)
H	int(short)
I	int(int,int)
J	int(short)
K	int(short,short)

===== Function B Symbol Table =====
return	int
a	int	param
x	int

===== Function C Symbol Table =====
return	int
a	int	param

===== Function D Symbol Table =====
return	int
a	int	param

===== Function E Symbol Table =====
return	int
a	int	param

===== Function J Symbol Table =====
return	int
a	short	param
d	double
b	double

===== Function K Symbol Table =====
return	int
a	short	param
b	short	param
o	int
p	int

Program
..FuncDeclaration
....Void
....Id(A)
....ParamList
......ParamDeclaration
........Void
..FuncDefinition
....Void
....Id(A)
....ParamList
......ParamDeclaration
........Void
........Id(a)
....FuncBody
......Declaration
........Int
........Id(x)
..FuncDefinition
....Int
....Id(B)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
......Declaration
........Int
........Id(x)
......Store - int
........Id(x) - int
........IntLit(2) - int
......Store - int
........Id(a) - int
........IntLit(3) - int
..FuncDeclaration
....Int
....Id(B)
....ParamList
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(C)
....ParamList
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(C)
....ParamList
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(C)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDeclaration
....Int
....Id(D)
....ParamList
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(D)
....ParamList
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(D)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDefinition
....Int
....Id(D)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDeclaration
....Int
....Id(E)
....ParamList
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(E)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDefinition
....Int
....Id(E)
....ParamList
......ParamDeclaration
........Double
........Id(b)
....FuncBody
..FuncDefinition
....Double
....Id(E)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDeclaration
....Int
....Id(F)
....ParamList
......ParamDeclaration
........Double
..FuncDefinition
....Int
....Id(F)
....ParamList
......ParamDeclaration
........Int
........Id(a)
....FuncBody
..FuncDeclaration
....Int
....Id(G)
....ParamList
......ParamDeclaration
........Short
........Id(a)
..FuncDefinition
....Short
....Id(G)
....ParamList
......ParamDeclaration
........Int
....FuncBody
......Declaration
........Int
........Id(y)
......Declaration
........Int
........Id(v)
......Return
........Id(v)
..FuncDeclaration
....Int
....Id(H)
....ParamList
......ParamDeclaration
........Short
........Id(a)
..FuncDeclaration
....Int
....Id(H)
....ParamList
......ParamDeclaration
........Short
........Id(a)
......ParamDeclaration
........Short
........Id(b)
..FuncDeclaration
....Int
....Id(I)
....ParamList
......ParamDeclaration
........Int
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(I)
....ParamList
......ParamDeclaration
........Int
..FuncDefinition
....Int
....Id(J)
....ParamList
......ParamDeclaration
........Short
........Id(a)
....FuncBody
......Declaration
........Double
........Id(d)
......Declaration
........Double
........Id(b)
......Return
........Id(b) - double
..FuncDefinition
....Int
....Id(J)
....ParamList
......ParamDeclaration
........Short
........Id(a)
......ParamDeclaration
........Short
........Id(b)
....FuncBody
......Declaration
........Int
........Id(a)
......Declaration
........Int
........Id(b)
......Return
........Id(b) - short
..FuncDefinition
....Int
....Id(K)
....ParamList
......ParamDeclaration
........Short
........Id(a)
......ParamDeclaration
........Short
........Id(b)
....FuncBody
......Declaration
........Int
........Id(o)
......Declaration
........Int
........Id(p)
......Return
........Id(p) - int
..FuncDefinition
....Int
....Id(K)
....ParamList
......ParamDeclaration
........Short
........Id(a)
....FuncBody
......Declaration
........Int
........Id(t)
......Declaration
........Int
........Id(r)
......Return
........Id(t) - int
